### PR TITLE
MVP Implementation

### LINKED ISSUE #2: Implement Move Mutation without validation
Mutations:

```graphql
move(
	input!: {
		player, 
		boardPosition!:Point, 
		tilePosition!:Point}
	){
		valid: Boolean
		message: string
		move: Move
	}

```

### LINKED ISSUE #4: Query

### PR DESCRIPTION
**Pull request by RohrerF**
resolves #4 
resolves #2

TODOs:
- [x] read configs from Environment Variables if running in container
- [x] tests for most important parts
	- [x] GameManager_Move_FirstMove_CorrectMoveNumber
	- [x] GameManager_Move_NextMove_CorrectMoveNumber
	- [x] GameManager_Move_ValidMove_SavesToMoveRepository
	- [x] GameManager_Move_ResultCorrect
- [x] finish #1 to be able to push container

### CODE REVIEW

File: src/Domain/GameManager.cs [hunk 1]
```diff
@@ -21,12 +21,7 @@ public GameManager(IGameRepository gameRepository, IMoveRepository moveRepositor
 
         public async Task<Game> CreateGame(CancellationToken cancellationToken)
         {
-            Game g = new Game
-            {
-                Id = Guid.NewGuid(),
-                Winner = null,
-                FinishedAt = null,
-            };
+            Game g = new Game {Id = Guid.NewGuid(), Winner = null, FinishedAt = null,};
```
Comment (panmona) [inline src/Domain/GameManager.cs]: Should we use something that is shorter than Guid that a human a can type in a reasonable time?
Something like: PPBqWA9 (https://github.com/dylang/shortid, this is not in C# though)

Comment (RohrerF) [reply src/Domain/GameManager.cs]: Yes later. Created Issue #7 for that

**Comment by RohrerF** [comment]
closes #1

File: src/Abstractions/Game.cs [hunk 2]
```diff
@@ -4,7 +4,7 @@ namespace UltimateTicTacToe.Abstractions
 {
     public class Game
     {
-        public Guid Id { get; set; }
+        public string Id { get; set; }
         public Player? Winner { get; set; }
```
Comment (panmona) [inline src/Abstractions/Game.cs:8]: So: 
draw and not finished: winner = null.
but with a draw the `FinishedAt` date is set aswell?

File: src/Domain.Abstractions/IGameManager.cs [hunk 3]
```diff
@@ -8,7 +7,13 @@ namespace UltimateTicTacToe.Domain.Abstractions
     public interface IGameManager
     {
         Task<Game> CreateGame(CancellationToken cancellationToken);
-        Task<bool> Move(Move m, CancellationToken cancellationToken);
+        Task<MoveResult> Move(Move m, CancellationToken cancellationToken);
     }
 
+    public class MoveResult
```
Comment (panmona) [inline src/Domain.Abstractions/IGameManager.cs]: Wouldn't it be better to put this into it's own file?

File: src/Api/Program.cs [hunk 4]
```diff
@@ -26,7 +25,10 @@ public static IWebHost BuildWebHost(string[] args)
             IConfigurationBuilder builder)
         {
             builder.SetBasePath(Directory.GetCurrentDirectory());
-            builder.AddJsonFile($"appsettings.json", optional: false, reloadOnChange: true);
+#if DEBUG
+            builder.AddJsonFile($"appsettings.json", optional: true, reloadOnChange: true);
```
Comment (panmona) [inline src/Api/Program.cs:29]: Why don't we need the appsettings.json when not running in DEBUG mode?

File: src/Domain/GameManager.cs [hunk 5]
```diff
@@ -15,37 +15,43 @@ public class GameManager : IGameManager
...
+            if (isValidMove)
+            {
+                await _moveRepository.Save(m, cancellationToken);
+            }
 
-            return await IsValidMove(m,cancellationToken);
+            return new MoveResult
+            {
+                IsValid = isValidMove,
+                Message = "", // TODO: make some nice messages here
```
Comment (panmona) [inline src/Domain/GameManager.cs]: Why do we need a message?

File: src/Abstractions/Game.cs (hunk 2 above)
Comment (RohrerF) [reply src/Abstractions/Game.cs:8]: the idea would be to do it that way. Made a note in #3 to avoid forgetting it.

File: src/Api/Program.cs (hunk 4 above)
Comment (RohrerF) [reply src/Api/Program.cs:29]: everything needs to be set in the Env-Variables. The appsettting file is not accessible in the docker image.

File: src/Domain/GameManager.cs (hunk 5 above)
Comment (RohrerF) [reply src/Domain/GameManager.cs]: i dont know. removed it

File: src/Api/Program.cs (hunk 4 above)
Comment (panmona) [reply src/Api/Program.cs:29]: You could mount it but that's fine for me.

**Review by panmona — APPROVED** [review]

**Pull request merged**